#Part 1: CONCEPTUAL QUESTIONS

1. What differs data science from data analytics?

Data science encompasses a broader scope, involving complex data processing, predictive modeling, and the creation of algorithms. Data analytics focuses on examining datasets to derive insights and support decision-making through analysis and visualization.

Data science creates tools and methods; data analytics applies them to specific problems.


2. Between mean and median, what do you think is more sensitive to outliers? Why?

The mean is more sensitive to outliers than the median because it considers all values in the dataset, including extreme ones, which can skew the average significantly. The median represents the middle value and remains relatively unaffected by outliers.

3. Is correlation also causation? Why or why not?

No. Correlation indicates a relationship/association between two variables, meaning they tend to move together in some way, but it does not prove that one variable causes the other. There may be other underlying factors or variables influencing both correlated variables, or the observed correlation could be coincidental.

#Part 2: Correlation Coefficients

Interpret the following in terms of mean, std, and correlation coefficient:

In [ ]:
import numpy as np

# Manually create a dataset with 15 observations
coffee_consumption = [2, 4, 1, 5, 3, 1, 3, 2, 4, 5, 1, 4, 3, 2, 5]
productivity_scores = [60, 75, 50, 80, 70, 55, 65, 60, 72, 85, 48, 78, 68, 58, 90]

# Calculate the mean and standard deviation for both arrays
mean_coffee = np.mean(coffee_consumption)
std_coffee = np.std(coffee_consumption)
mean_productivity = np.mean(productivity_scores)
std_productivity = np.std(productivity_scores)

# Calculate the correlation coefficient between coffee consumption and productivity scores
# 2 and 60, 4 and 75, 1 and 50...5 and 90 then calculates how closely paired values align on a linear scale
# This coefficient is a single number that represents overall relationship between consumption and productivity scores in dataset
correlation_coefficient = np.corrcoef(coffee_consumption, productivity_scores)[0, 1]

(mean_coffee, std_coffee, mean_productivity, std_productivity, correlation_coefficient)

# On average coffee consumption in dataset is 3 - individuals in dataset consume 3 units of coffee
# std is 1.41 for coffee. It seems high compared to mean of 3 - variability to coffee consumption amongst individuals in dataset?
# On average productivity (mean) of 67.6
# std of productivity is 12; Most scores lie around 12 points above or below mean of 67.6; seems high
# productivity ranges 48-90 w/15 in array (42 points) With std of 12 there's considerable variability

(3.0, 1.4142135623730951, 67.6, 12.147976511885973, 0.9740102362132356)

[INTERPRETATION HERE]

In [ ]:
import numpy as np

# Manually create a dataset with 15 observations
hours_of_tv = [2, 3, 2, 4, 5, 1, 3, 2, 4, 5, 1, 4, 3, 2, 5]
grades = [78, 80, 75, 82, 85, 70, 79, 77, 81, 88, 69, 84, 79, 76, 87]

# Calculate the mean and standard deviation for both arrays
mean_hours_of_tv = np.mean(hours_of_tv)
std_hours_of_tv = np.std(hours_of_tv)
mean_grades = np.mean(grades)
std_grades = np.std(grades)

# Calculate the correlation coefficient between hours of tv and grades
correlation_coefficient = np.corrcoef(hours_of_tv, grades)[0, 1]

(mean_hours_of_tv, std_hours_of_tv, mean_grades, std_grades, correlation_coefficient)

(3.066666666666667,
 1.339983416149452,
 79.33333333333333,
 5.3499740393970345,
 0.9640438248749076)

Average of 3.07 hours of TV watching with standard deviation of 1.34 hours.

Average grade around 79.33 with standard deviation of 5.35.

High correlation of 0.964 a strong positive linear relationship between hours of TV watched and grades achieved. Yay, increased hours of TV watched - increased grades. This is because 0.964 is close to 1.

[INTERPRETATION HERE]

In [ ]:
import numpy as np

# Manually create a dataset with 15 observations
x = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]
y = [2, 4, 6, 8, 45, 12, 14, 45, 18, 20, 22, 24, 46, 28, 45]

# Calculate the mean and standard deviation for both arrays
mean_x = np.mean(x)
std_x = np.std(x)
mean_y = np.mean(y)
std_y = np.std(y)

# Calculate the correlation coefficient between hours of tv and grades
correlation_coefficient = np.corrcoef(x, y)[0, 1]

(mean_x, std_x, mean_y, std_y, correlation_coefficient)

(8.0, 4.320493798938574, 22.6, 15.400432894348565, 0.6612814826286639)

x: 8 mean, 4.32 std; y: 22.6 mean, 15.4 std; corr coeff: 0.661

Moderate positive linear relationship between x and y; as x increases, y tend to increase with some variability.

[INTERPRETATION HERE]

#Part 3: Probability

1. Find the probability of getting a numbered card when a card is drawn from the pack of 52 cards.

Probability of an event happening = Number of ways it can happen / Total number of outcomes

OR

Probability = Number of numbered cards / Total numbers of cards

36/52 = .6923 = **69.23%**

2. You have a bag containing 8 red balls, 5 green balls, and 3 blue balls. If you randomly draw two balls without replacement, what is the probability that both balls are green?

Total number of balls = 8 red + 5 green + 3 blue = 16;

Probability of green ball on first draw is 5/16;

Probability of green ball on 2nd draw provided green ball drawn on first draw is 4/15 (4 green left, 15 total left after first draw).

Probability = 5/16 X 4/15 = 20/240 = 1/12 = .0833 = **8.33%**

3. You roll a six-sided die. What is the probability of rolling an even number or a number greater than 4?

6 sides (possible outcomes); 3 even numbers (2,4,6); 3 odd numbers (1,3,5); Numbers > 4 (5, 6). Don't count 6 twice (even and >4).

Probability = Favorable outcomes / Total outcomes

3 + 2 -1 / 6 (3 even numbers, 2 numbers > 4, -1 so we don't count 6 twice)
4/6 = .6666 = **66.67%**